### PR TITLE
use addInefficientCastReason everywhere possible

### PR DESCRIPTION
**Pull request by ToppleTheNun**
### Description

Use `addInefficientCastReason` everywhere instead of `event.meta.isInefficientCast` so that logic can be centralized.

Adds an ESLint rule for enforcing this.

### CODE REVIEW

**Comment by ToppleTheNun** [comment]
@emallson need your eyes on this section in Blood: https://github.com/ToppleTheNun/WoWAnalyzer/blob/4797b518ccc62ae6a9ed4dacecbc3876193b19b6/src/analysis/retail/deathknight/blood/components/ResourceWasteProblemRenderer.tsx#L48

File: src/analysis/retail/deathknight/blood/modules/features/Ossuary.tsx [hunk 1]
```diff
@@ -40,12 +46,7 @@ class Ossuary extends Analyzer {
     } else {
       this.dsWithoutOS += 1;
 
-      event.meta = event.meta || {};
-      event.meta.isInefficientCast = true;
-      event.meta.inefficientCastReason = defineMessage({
-        id: 'deathknight.blood.ossuary.ineffectiveCast',
-        message: `This Death Strike cast was without Ossuary.`,
-      });
+      addInefficientCastReason(event, ineffectiveCastMessage);
```
Comment (emallson) [inline src/analysis/retail/deathknight/blood/modules/features/Ossuary.tsx:49]: this one will conflict with my blood apl pr because its actually broken and displays `[Object object]` right now :see_no_evil: will be an easy conflict to fix though. will revert the change in my pr because this solution is better.

File: src/analysis/retail/deathknight/blood/components/ResourceWasteProblemRenderer.tsx [hunk 2]
```diff
@@ -45,6 +45,8 @@ export function ResourceWasteProblemRenderer({
               return <>This cast wasted {this._wastedRp} RP.</>;
             },
           };
+          // TODO: Come up with a better way of handling this so that it doesn't break any existing meta
+          // eslint-disable-next-line wowanalyzer/event-meta-inefficient-cast
```
Comment (emallson) [inline src/analysis/retail/deathknight/blood/components/ResourceWasteProblemRenderer.tsx]: so this is operating on a copy of events and intentionally overwriting the meta of the copy (see [this branch](https://github.com/WoWAnalyzer/WoWAnalyzer/pull/6783/files#diff-710f818e171abb0cb5e6daff3b65c2986efefa10f915e06f49dd55e04dd36854R57-R62)). we could potentially add a `replace` option, or just allow this to slide with an `eslint-disable` like you have.

**Comment by emallson** [comment]
the overall approach looks good IMO. just a question of what to do with my blood shenanigans :sweat_smile:

File: src/analysis/retail/deathknight/blood/components/ResourceWasteProblemRenderer.tsx (hunk 2 above)
Comment (ToppleTheNun) [reply src/analysis/retail/deathknight/blood/components/ResourceWasteProblemRenderer.tsx]: I'll write a `replace` function in EventMetaLib. The other thing I've been thinking about is getting the type of the EventMeta off of the event, which could have a function for it here as well.

**Review by emallson — APPROVED** [review]
